sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalog.v2.{Identifier, TableCatalog}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, GenericRowWithSchema}
+import org.apache.spark.sql.catalyst.plans.DescribeTableSchemas
+import org.apache.spark.sql.execution.LeafExecNode
+import org.apache.spark.sql.sources.v2.Table
+
+case class DescribeTableExec(
+    catalog: TableCatalog,
+    ident: Identifier,
+    isExtended: Boolean) extends LeafExecNode {
+
+  import DescribeTableExec._
+
+  override def output: Seq[AttributeReference] = DescribeTableSchemas.DESCRIBE_TABLE_ATTRIBUTES
+
+  override protected def doExecute(): RDD[InternalRow] = {
+    val rows = new ArrayBuffer[InternalRow]()
+    if (catalog.tableExists(ident)) {
+      val table = catalog.loadTable(ident)
+      addSchema(rows, table)
+
+      if (isExtended) {
+        addPartitioning(rows, table)
+        addProperties(rows, table)
+      }
+
+    } else {
+      rows += toCatalystRow(s"Table $ident does not exist.", "", "")
```

Comment:
shouldn't we throw exception when table not found?

Reply:
I think we can follow the https://github.com/apache/spark/pull/24937: The `DescribeTable` should contain an `UnresolvedRelation`, so that analyzer can check table existence for us.

Reply:
Followed the `AlterTable` approach in the latest commit.

Reply:
Ah but I didn't remove this - though I guess technically we should never hit this code path. We can throw an exception here instead.